packages/esm-appointments-app/src/appointments/scheduled/scheduled-appointments.component.tsx

```diff
@@ -16,101 +23,165 @@ interface ScheduledAppointmentsProps {
 
 type DateType = 'pastDate' | 'today' | 'futureDate';
 
-const ScheduledAppointments: React.FC<ScheduledAppointmentsProps> = ({ visits, appointmentServiceType }) => {
-  const scheduledAppointmentsPanelsSlot = 'scheduled-appointments-panels-slot';
+const scheduledAppointmentsPanelsSlot = 'scheduled-appointments-panels-slot';
 
+const ScheduledAppointments: React.FC<ScheduledAppointmentsProps> = ({ visits, appointmentServiceType }) => {
   const { t } = useTranslation();
   const { currentAppointmentDate: date } = useAppointmentDate();
 
   const [currentTab, setCurrentTab] = useState<string>(null);
   const [dateType, setDateType] = useState<DateType>('today');
   const scheduledAppointmentPanels = useConnectedExtensions(scheduledAppointmentsPanelsSlot);
-
-  const isDateInPast = dayjs(date).isBefore(dayjs(), 'date');
-  const isDateInFuture = dayjs(date).isAfter(dayjs(), 'date');
-  const isToday = dayjs(date).isSame(dayjs(), 'date');
+  const { allowedExtensions, showExtension, hideExtension } = useAllowedExtensions();
+  const shouldShowPanel = useCallback(
```

Comment:
This is just a filtering function that's used to control which tabs get rendered.